chronos/asyncfutures2.nim

```diff
@@ -397,75 +448,82 @@ proc mget*[T](future: FutureVar[T]): var T =
   ## Future has not been finished.
   result = Future[T](future).value
 
-proc finished*(future: FutureBase | FutureVar): bool =
-  ## Determines whether ``future`` has completed.
-  ##
-  ## ``True`` may indicate an error or a value. Use ``failed`` to distinguish.
-  when future is FutureVar:
-    result = (FutureBase(future)).finished
-  else:
-    result = future.finished
-
-proc failed*(future: FutureBase): bool =
-  ## Determines whether ``future`` completed with an error.
-  return (not isNil(future.error))
-
 proc asyncCheck*[T](future: Future[T]) =
   ## Sets a callback on ``future`` which raises an exception if the future
   ## finished with an error.
   ##
   ## This should be used instead of ``discard`` to discard void futures.
   doAssert(not isNil(future), "Future is nil")
   proc cb(data: pointer) =
-    if future.failed:
+    if future.failed() or future.cancelled():
```

Comment:
Absolutely agree with deprecation of `asyncCheck`, but could not agree with treating `cancellation` not as an error. Cancellation is not a normal code flow, its exceptional way of doing things to avoid leaks and other problems.

Reply:
Well in my world cancellation is a perfectly normal codeflow. Also it's a neat natural way to break spawned loops. But thinking further about this particular case, I conclude that it's nearly impossible to hit `cancelled` condition in `asyncCheck`, because usually `asyncCheck` implies that you don't have the reference to the future, and thus will not be able to cancel it, so I no longer care :)